demes/demes.py

```diff
@@ -519,12 +510,20 @@ def __attrs_post_init__(self):
                 raise ValueError("ancestors and proportions must have same length")
             if self.id in self.ancestors:
                 raise ValueError(f"{self.id} cannot be its own ancestor")
+        # every deme must have at least one epoch
```

Comment:
Should these be a @epochs.validator decorated function, since we're checking properties of the `epochs` list we passed to `Deme`?

Reply:
I don't think it makes any difference either way. So it's up to you as to whether you think it's cleaner to have one validation function, or more than one.

Reply:
Gotcha. Ok, going to leave as here and merge when checks pass